### PR TITLE
Add initial test plan for media stress tests

### PR DESCRIPTION
**Pull request by mckees**

### CODE REVIEW

File: checkbox-provider-kivu/bin/gpu-load.py [hunk 1]
```diff
@@ -29,22 +29,26 @@ def compute_avg_intel(fname : str) -> float:
...
         gpu_average = 0.0
         sumdata = {}
         for d in data:
+            # Get the duration of the measurement
+            duration_tuple, _ = d["period"].items()
             # Looking at the `engines` section only
             for k, v in d["engines"].items():
                 # Focus on GPU usage for video encoding/decoding only
-                if "Video/" in k and (v.get("busy") is not None):
+                if "Video/" in k and (v.get("busy") is not None) and v["busy"] > threshold:
```
Comment (petitfou) [inline checkbox-provider-kivu/bin/gpu-load.py:41]: where does the variable threshold come from ?

File: checkbox-provider-kivu/bin/gpu-load.py [hunk 2]
```diff
@@ -29,22 +29,26 @@ def compute_avg_intel(fname : str) -> float:
         else:
             data = json.loads(content)
```
Comment (petitfou) [inline checkbox-provider-kivu/bin/gpu-load.py:30]: The semantic of the module and returned value now change a little bit depending on the argument, i would prefer an other script for that, but if we still want to keep only one file, there will be a need to document it a little bit

File: checkbox-provider-kivu/units/jobs.pxu [hunk 3]
```diff
@@ -493,3 +493,44 @@ command:
...
+  echo "{\"gpu_usage_{{ driver }}\": $(sudo ${GPU_LOAD_CMD} --timeout=30 --threshold=99.3 --gpu={{ driver }})}" | tdb.py insert &
+  PLAY_TIMEOUT=12
+  # Allows a unique variable to be declared per-process
+  for i in {1..20}; do
+  exec sudo --preserve-env -u "${NORMAL_USER}" timeout 30 mpv --loop -no-audio --geometry=20% "${PLAINBOX_PROVIDER_DATA}"/bbb_h264_2160p_60fps_extract.mp4 &
+  done
+  sleep ${PLAY_TIMEOUT}
+  tdb.py wait
+  # Checks the number of ms spent above the threshold for the video engine decoder
+  # The stress test uses both Video/0 and Video/1, so this really checks for 24s
```
Comment (petitfou) [inline checkbox-provider-kivu/units/jobs.pxu:533]: 24s is not too hard and will make the test fail from times to times ?

**Review by petitfou — CHANGES_REQUESTED** [review]

**Comment by mckees** [comment]
This version also has an issue with single-engine machines (My coffee lake system does not have Video/0 and Video/1, so simply dividing by two to get the time spent above the video engine utilization threshold is not a good solution. Uploading a fix for that in the next series

**Comment by mckees** [comment]
@petitfou the main addition to the new patches is to add the function which counts video engines so that we don't have issues between machines that have a different number of them. If the extra code in the gpu-load script doesn't sit well with you, I can refactor out the common code like I mentioned in my inline comment above

File: checkbox-provider-kivu/units/jobs.pxu [hunk 4]
```diff
@@ -490,3 +490,42 @@ command:
...
+environ:
+  # necessary for local mode
+  XDG_SESSION_TYPE
+  XDG_RUNTIME_DIR
+  NORMAL_USER
+command:
+  tdb.py reset
+  GPU_LOAD_CMD=$(which gpu-thresh.py)
+  # Since it's a stress test, set the threshold near 100% video engine utilization
+  echo "{\"gpu_usage_{{ driver }}\": $(sudo ${GPU_LOAD_CMD} --timeout=30 --video_engine=Video --threshold=99.3 --gpu={{ driver }})}" | tdb.py insert &
```
Comment (petitfou) [inline checkbox-provider-kivu/units/jobs.pxu:521]: rename gpu_usage to something elso (gpu_usage_over_threshold_duration ?)

File: checkbox-provider-kivu/units/jobs.pxu [hunk 5]
```diff
@@ -490,3 +490,42 @@ command:
...
+unit: template
+template-resource: graphics_card
+template-filter: graphics_card.driver in ['i915','amdgpu','amdgpu-pro']
+template-engine: jinja2
+template-unit: job
+id: kivu/stress_mpv_h264_decoding_{{ driver }}
+category_id: kivu
+flags: simple
+user: root
+_summary: Play many H264 videos (VAAPI enabled) as a stress test and capture GPU usage
```
Comment (petitfou) [inline checkbox-provider-kivu/units/jobs.pxu:502]: Modify summary to better match the test

File: checkbox-provider-kivu/bin/gpu-thresh.py [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        "--gpu", type=str, help="GPU driver (amdgpu, i915)"
+    )
+    parser.add_argument(
+        "--file", type=str, help="Stat output file"
+    )
+    parser.add_argument(
+        "--threshold", type=float, help="Minimum threshold for video engine utilization"
+    )
+    parser.add_argument(
+        "--video_engine", type=str, help="Manually set the video engine you" \
```
Comment (petitfou) [inline checkbox-provider-kivu/bin/gpu-thresh.py:59]: you can use choices=['Video', 'VideoEnhance'], default='Video' , with that, you will be able to remove the check later on ...

File: checkbox-provider-kivu/bin/gpu-thresh.py [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+                " want to monitor (Video (Default) | VideoEnhance"
+    )
+    args = parser.parse_args()
+
+    timeout = 1 if args.timeout is None else args.timeout
+    gpu_driver = 'i915' if args.gpu is None else args.gpu
+    fname = '/tmp/gpu-thresh-5ef38178-c2c0-11ed-afa1-0242ac120002.data' if args.file is None else args.file
+    threshold = 0 if args.threshold is None else args.threshold
+
+    if args.video_engine is None:
```
Comment (petitfou) [inline checkbox-provider-kivu/bin/gpu-thresh.py:69]: ... here , juste have to do : video_engine = f'{args.video_engine}/'

**Review by petitfou — CHANGES_REQUESTED** [review]

**Review by petitfou — APPROVED** [review]
LGTM +1

**Comment by mckees** [comment]
@petitfou I don't have merge rights. Could you please merge for me?

**Pull request merged**